classes/statistics/PKPTemporaryTotalsDAO.php

```diff
@@ -108,7 +108,11 @@ public function removeDoubleClicks(int $counterDoubleClickTimeFilter): void
     public function compileContextMetrics(string $loadId): void
     {
         $date = substr($loadId, -12, 8);
-        DB::table('metrics_context')->where('load_id', '=', $loadId)->orWhere('date', '=', DB::raw("DATE({$date})"))->delete();
+        $dateFormatSql = "STR_TO_DATE('{$date}', '%Y%m%d')";
+        if (substr(Config::getVar('database', 'driver'), 0, strlen('postgres')) === 'postgres') {
```

Comment:
Rather than a loose test on the driver name, I'd suggest checking the class name of the driver, e.g.:
https://github.com/pkp/pkp-lib/blob/stable-3_4_0/classes/migration/upgrade/PKPv3_3_0UpgradeMigration.php#L155

But better still -- since we have the date string in PHP-land, why not use something like https://www.php.net/manual/en/datetimeimmutable.createfromformat.php to parse the string into a saner format, then use e.g. Laravel's `whereDate` to pass into the database regardless of which DBMS is used?